src/main/java/com/sourcegraph/javagraph/SbtProject.java

```diff
@@ -0,0 +1,190 @@
+        }
+    }
+
+    private static Collection<SourceUnit> processProject(Path path, String project, BufferedReader reader, BufferedWriter writer) throws IOException {
+        executeSbtCommand("project " + project, reader, writer);
+
+        Collection<String> subprojects = executeSbtCommand("show name", reader, writer);
+        // project may include subprojects which we need to parse
+        if (subprojects.size() > 1) {
+            Collection<SourceUnit> sourceUnits = new ArrayList<>();
+            for (String subproject : subprojects) {
+                String prefix = "[info] ";
+                if (!subproject.startsWith(prefix)) {
+                    continue;
+                }
+                subproject = subproject.substring(prefix.length());
+                if (subproject.startsWith("    ")) {
+                    continue;
+                }
+                String[] data = subproject.split("/", 2);
+                project = data[0];
+                if (project.equals(subproject)) {
+                    continue;
+                }
+                LOGGER.debug("found subproject: {}", subproject);
+                sourceUnits.addAll(processProject(path, subproject, reader, writer));
+            }
+            return sourceUnits;
+        }
+
+        // only a single project
+        SourceUnit unit = new SourceUnit();
+
+        unit.Name = subprojects.iterator().next();
+        unit.Type = "ScalaArtifact";
+
+        // parse dependencies
+        for (String dependency : executeSbtCommand("show libraryDependencies", reader, writer)) {
+            String prefix = "* ";
+            Preconditions.checkState(dependency.startsWith(prefix), "unexpected dependency prefix: %s", dependency);
+            dependency = dependency.substring(prefix.length());
+            if (dependency.length() == 0) {
+                continue;
+            }
+
+            String[] dep = dependency.split(":");
+            Preconditions.checkState(dep.length >= 3, "unexpected dependency format: %s", dependency);
+            RawDependency raw = new RawDependency(/*groupID=*/ dep[0],
+                    /*artifactID=*/ dep[1], /*version=*/ dep[2],
+                    /*scope=*/ dep.length >= 4 ? dep[3] : null, /*file=*/ null,
+                    PathUtil.relativizeCwd(path.toAbsolutePath()).toString());
+            raw.type = "jar";
+            unit.Dependencies.add(raw);
+        }
+
+        // parse source files
+        unit.Files = new ArrayList<>();
+        for (String sourceFile : executeSbtCommand("show unmanagedSources", reader, writer)) {
+            String prefix = "* ";
+            // ignore trailer: "[success] Total time: 0 s, completed Oct 19, 2017 2:16:57 PM"
+            if (sourceFile.startsWith(prefix)) {
+                unit.Files.add(sourceFile.substring(prefix.length()));
+            }
+        }
+
+        executeSbtCommand("show version", reader, writer);  // TODO: SourceUnit lacks a field for version
```

Comment:
are we handling this somewhere?

Reply:
Removed.  In gradle builds I found a few things with a `SRCLIB-` prefix which are not consumed.